### DISCUSSION TITLE
[v7] Apply variantColorResolver for all components that include filled (at least) variant

### DISCUSSION THREAD

**Discussion by 7iomka** [mantinedev/mantine#5070] [Feature requests]
### What package has an issue

@mantine/core

### Describe the bug

Now v7 is applied for all button-like components.
But Tabs for example with variants that uses background for active tab (pills for example) doesn't don't receive the same setiings which I would like to se applied.
For example my primary color is very light, and every mantine component that uses filled variant I need now to change somehow to make content/text color black instead of hardcoded in the css-variable white color.
With variant color resolver I hust used

```ts
  // Override primary text color
  if (isPrimaryColor && input.variant === 'filled') {
    return { ...defaultResolvedColors, color: 'var(--dark-400)' };
  }
```

but for Tabs I need to use another approach in `vars`
```ts
  const parsedColor = parseThemeColor({
    color: color || theme.primaryColor,
    theme,
  });

  const resolvedColors = defaultVariantColorsResolver({
    variant: 'filled',
    color: parsedColor.color,
    theme,
  });



  ---tabs-text-color: parsedColor.color.includes('primary') ? 'black' : 'white'

```

then to use it in css styles

For `SegmentedControl` I need to do the same, but the process is even harder because we not able to set `--sc-label-color` with vars on Provider level.

### What version of @mantine/* packages do you have in package.json? (Note that all @mantine/* packages must have the same version in order to work correctly)

latest

### Do you know how to fix the issue

None

### Are you willing to participate in fixing this issue and create a pull request with the fix

None

**Comment by rtivital** [comment]
It is not feasible. You can define custom CSS variables instead with CSS variables resolver – https://mantine.dev/styles/css-variables/#css-variables-resolver

**Comment by 7iomka** [comment]
> It is not feasible. You can define custom CSS variables instead with CSS variables resolver – https://mantine.dev/styles/css-variables/#css-variables-resolver

Can you share the example how to do this for SegmentedControl for example?
```
    --sc-label-color: var(--mantine-color-white);
```
I don't really understand why this value is hardcoded if the original component styles have a fallback value? I would expect this variable on the root element and could easily override it
